### PR TITLE
feat: explain polkadotjs

### PR DESCRIPTION
**Pull request by trusch**
## fixes KILTProtocol/ticket#2410
* add two short paragraphs about why polkadotjs is used
  * We mainly use it in the docs about the linking
* removed one unneccessary usage of it

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ As a DID can have multiple keys, in addition to the signature data itself, a DID
 
 The snippet below shows how to generate and verify a DID signature using the KILT SDK.
 
+You may notice the import of the `u8aToHex` utility function from the `@polkadot/util` package. This is because the KILT SDK uses the [polkadot.js](https://polkadot.js.org/docs/) libraries, that are intended to be used to interact with [substrate](https://docs.substrate.io/) based blockchains like KILT.
+Those libraries provide all the necessary basics, like cryptographic primitives or serialization, but they do not provide any high-level functionality.
```
Comment (weichweich) [inline docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md]: I like it. But please remove the import of `u8aToHex` with `Kilt.Utils.Crypto.u8aToHex`. :grin: Since it's exported from the SDK we shouldn't use the polkadot util for that.

File: code_examples/sdk_examples/src/core_features/web3names/04_release.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import * as Kilt from '@kiltprotocol/sdk-js'
 
 export async function releaseWeb3Name(
   did: Kilt.DidUri,
-  submitterAccount: Kilt.KiltKeyringPair,
+  submitterKeyPair: Kilt.KiltKeyringPair,
```
Comment (weichweich) [inline code_examples/sdk_examples/src/core_features/web3names/04_release.ts]: What's the reason for this change? It's called `submitterAccount` to distinguish it from the DID and emphasize that it's the keypair of an account.

**Review by weichweich — CHANGES_REQUESTED** [review]

File: code_examples/sdk_examples/src/core_features/web3names/04_release.ts (hunk 2 above)
Comment (trusch) [reply code_examples/sdk_examples/src/core_features/web3names/04_release.ts]: I can revert this, but I felt it that name would be clearer, since this is not only account info, but also its private key. But I see that we should be consistent with this, since we also use "submitterAccount" often elsewhere.

Comment (weichweich) [reply code_examples/sdk_examples/src/core_features/web3names/04_release.ts]: Yeah either rename it everywhere or don't. I'm still in favour of not renaming it. But I also see the argument for calling it `submitterKeyPair`. So it's up to you...

File: docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ As a DID can have multiple keys, in addition to the signature data itself, a DID
 
 The snippet below shows how to generate and verify a DID signature using the KILT SDK.
 
+You may notice the import of the `u8aToHex` utility function from the `@polkadot/util` package. This is because the KILT SDK uses the [polkadot.js](https://polkadot.js.org/docs/) libraries, that are intended to be used to interact with [substrate](https://docs.substrate.io/) based blockchains like KILT.
```
Comment (ntn-x2) [inline docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md]:
```suggestion
You may notice the import of the `u8aToHex` utility function from the `@polkadot/util` package. This is because the KILT SDK uses the [polkadot.js](https://polkadot.js.org/docs/) libraries, that are intended to be used to interact with [Substrate](https://docs.substrate.io/)-based blockchains like KILT.
```

File: docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md (hunk 1 above)
Comment (ntn-x2) [reply docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md]: If we remove that, this whole section needs a different reason as to why it's here, cause the way it was introduced was exactly because of `u8aToHex`.

**Review by ntn-x2 — CHANGES_REQUESTED** [review]
Unfortunately, I do not think these explanations are sufficient. We use `api.tx` and `api.call` everywhere, and albeit they are augmentations provided by the SDK, they still augment the basic `api` type. As I mentioned previously, I think we should have a basic definition + relevant link to the `api` object and `@polkadot/api` library in the glossary, and also in the other entry points into our docs, which I think for now it's the cookbook quickstart and the workshop. Both could point to the definition in the glossary.

File: docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md (hunk 1 above)
Comment (trusch) [reply docs/develop/01_sdk/02_cookbook/01_dids/07_did_signature.md]: reverted that completely.

File: code_examples/sdk_examples/src/core_features/web3names/04_release.ts (hunk 2 above)
Comment (trusch) [reply code_examples/sdk_examples/src/core_features/web3names/04_release.ts]: reverted.

File: docs/develop/03_workshop/04_attester/01_account.md [hunk 4]
```diff
@@ -39,6 +39,15 @@ A person can memorize it, and use it later to re-generate their keypairs and add
 To generate an account, we use the `addFromMnemonic()` function on the [`KiltKeyringPair`](https://kiltprotocol.github.io/sdk-js/interfaces/_kiltprotocol_types.KiltKeyringPair.html) generated via the SDK.
 The mnemonic is generated with the polkadot function `mnemonicGenerate()`, which generates a 12-word mnemonic.
 
+:::info polkadot.js
+
+The KILT SDK is built on top of the [polkadot.js](https://polkadot.js.org/) library therefore you will find various uses of it in this workshop.
+It provides a set of tools to interact with the KILT blockchain and other substrate based blockchains, provides cryptographic primitives and a serialization framework.
```
Comment (weichweich) [inline docs/develop/03_workshop/04_attester/01_account.md]: It was a bit hard to follow. 

My proposal:

```suggestion
It provides a set of tools to interact with the KILT blockchain and other substrate based blockchains
In addition the polkadot-js library offers cryptographic primitives and a serialization framework to decode data received from the blockchain.
```

ChatGPTs proposal:
```suggestion
This JavaScript library includes a collection of tools for interacting with the KILT blockchain and other Substrate-based blockchains. Moreover, it offers cryptographic primitives and a serialization framework to facilitate blockchain development.
```

**Review by weichweich — APPROVED** [review]
Looks okay. This one sentence need a rewrite though

File: docs/concepts/09_glossary.md [hunk 5]
```diff
@@ -143,6 +143,12 @@ More details about Parachains can be found at the [offical Polkadot documentatio
 A multi-chain network that allows for interoperability between different blockchain protocols, including the KILT Protocol.
 More details about Polkadot can be found on their [offical documentation](https://wiki.polkadot.network/docs/getting-started#what-is-polkadot)
 
+### polkadot.js
+
+A JavaScript library that allows developers to interact with the substrate based blockchains, including the KILT Protocol.
+It is the basis of the KILT SDK and provides many utils and functions that may be usefull for application developers.
```
Comment (ntn-x2) [inline docs/concepts/09_glossary.md]:
```suggestion
It is the basis of the KILT SDK and provides many utils and functions that may be useful for application developers.
```

File: docs/concepts/09_glossary.md [hunk 6]
```diff
@@ -143,6 +143,12 @@ More details about Parachains can be found at the [offical Polkadot documentatio
 A multi-chain network that allows for interoperability between different blockchain protocols, including the KILT Protocol.
 More details about Polkadot can be found on their [offical documentation](https://wiki.polkadot.network/docs/getting-started#what-is-polkadot)
 
+### polkadot.js
+
+A JavaScript library that allows developers to interact with the substrate based blockchains, including the KILT Protocol.
```
Comment (ntn-x2) [inline docs/concepts/09_glossary.md]:
```suggestion
A JavaScript library that allows developers to interact with Substrate-based blockchains, including KILT Spiritnet and Peregrine.
```

File: docs/develop/03_workshop/04_attester/01_account.md (hunk 4 above)
Comment (ntn-x2) [reply docs/develop/03_workshop/04_attester/01_account.md]: One-sentence-one-line tho:

```suggestion
This JavaScript library includes a collection of tools for interacting with the KILT blockchain and other Substrate-based blockchains.
Moreover, it offers cryptographic primitives and a serialization framework to facilitate blockchain development.
```

File: docs/develop/03_workshop/04_attester/01_account.md [hunk 7]
```diff
@@ -39,6 +39,15 @@ A person can memorize it, and use it later to re-generate their keypairs and add
...
+We recommend that you familiarize yourself with the [API documentation](https://polkadot.js.org/docs/) to learn more about the functions available and get a deeper understanding of how things work.
```
Comment (ntn-x2) [inline docs/develop/03_workshop/04_attester/01_account.md]: I would assume the intent is always to understand better how things work, so I would remove that section.

```suggestion
We recommend developers to familiarize themselves with their [API documentation](https://polkadot.js.org/docs/) to learn more about the functions available.
```

**Review by ntn-x2 — COMMENTED** [review]
It looks already better! But I think we should also add a small paragraph about polkadot.js (similar to what we do for the workshop) in the quickstart guide, since we use `api.query`... also there, and we should give a small heads-up to users as to what that `api` object actually is.
